index.js

```diff
@@ -380,17 +382,28 @@
     };
   }
 
+  function typeX(name, test, _1, _2) {
+    var version = '0.7.0';  // updated programmatically
+    var url = 'https://github.com/sanctuary-js/sanctuary-def/tree/v' +
+              version + '#' + stripNamespace(name);
+    switch (arguments.length) {
```

Comment:
What about something like this? (this names are not perfect)
```js
$.FiniteNumber = withURL('sanctuary-def/FiniteNumber', (name, url) => NullaryType(
  name, url,
  function(x) { return $.ValidNumber._test(x) && isFinite(x); }
));
// or 
$.FiniteNumber = typeWithURL(
  'sanctuary-def/FiniteNumber',
  NullaryType,
  function(x) { return $.ValidNumber._test(x) && isFinite(x); }
);
```
Whis way we would not need this switch statement. And it will eliminate need to update it if we add some other `*Type` with more arguments. Also it will still be clear which type a value is (user will scan the definition and see `NullaryType` or whatever immediately instead of counting number of arguments.


```js
$.Pair = typeWithURL(
  'sanctuary-def/Pair',
  BinaryType,
  function(x) { return $$typeEq('Array')(x) && x.length === 2; },
  function(pair) { return [pair[0]]; },
  function(pair) { return [pair[1]]; }
)
// vs
$.Pair = typeX(
  'sanctuary-def/Pair',
  function(x) { return $$typeEq('Array')(x) && x.length === 2; },
  function(pair) { return [pair[0]]; },
  function(pair) { return [pair[1]]; }
)
```